### PR TITLE
Added ability to grab number of ghosts and grimoire tally during lore…

### PR DESCRIPTION
**Pull request by JacobCallahan**
… prep

Also make some improvements to the lore, ghosts, and grim commands.

### CODE REVIEW

File: plugins/destiny.py [hunk 1]
```diff
@@ -89,14 +89,22 @@ def prepare_lore_cache():
...
-        for level2 in level1['pageCollection']:
-            for card in level2['cardCollection']:
+        for level2 in level1.get('pageCollection', []):
+            for card in level2.get('cardCollection', []):
                 LORE_CACHE[card['cardName']] = {
                     'cardIntro': card.get('cardIntro', ''),
                     'cardDescription': card['cardDescription'],
                     'cardId': card['cardId']
                 }
+                if 'Ghost Fragment' in card['cardName']:
```
Comment (Jessidhia) [inline plugins/destiny.py]: Not all Ghost cards have "Ghost Fragment" in their name. For example, [the new "Osiris: Vision 81" card](https://www.reddit.com/r/DestinyTheGame/comments/3ksqds/an_11th_dead_ghost_has_been_found_no_glitching/).

You could hardcode the amount of exceptions as the initial `ghost_count`, but we won't know how many there are until launch :X

File: plugins/destiny.py [hunk 2]
```diff
@@ -422,13 +430,15 @@ def lore(text, bot):
...
     else:
         matches = []
         for entry in LORE_CACHE:
-            if text.lower() in entry.lower():
+            if text.lower() == entry.lower():
                 name = entry
-            elif text.lower() in entry.lower():
-                matches.append(entry)
+            elif text.lower() in entry.lower() or text.lower() in LORE_CACHE[entry].get('cardDescription', '').lower():
+                    matches.append(entry)
```
Comment (Jessidhia) [inline plugins/destiny.py]: Too much indent.

File: plugins/destiny.py [hunk 3]
```diff
@@ -422,13 +430,15 @@ def lore(text, bot):
     name = ''
     if not text:  # if we aren't searching, return a random card
         name = sample(list(LORE_CACHE), 1)[0]
+        if name in {'ghost_count', 'grim_tally'}:
```
Comment (Jessidhia) [inline plugins/destiny.py]: `while`, not `if`. The `sample` call might be unlucky enough to get a "bad word" again.

Comment (Jessidhia) [inline plugins/destiny.py]: Hm, could also just sample 3 ("badwords" + 1) and shift the sampled list until you get a not-badword.

File: plugins/destiny.py (hunk 1 above)
Comment (JacobCallahan) [inline plugins/destiny.py]: once we know them all, we can adjust.

File: plugins/destiny.py (hunk 2 above)
Comment (JacobCallahan) [inline plugins/destiny.py]: yup sure is

File: plugins/destiny.py (hunk 3 above)
Comment (JacobCallahan) [inline plugins/destiny.py]: i like the while approach. good call

**Pull request merged**